packages/itwin/tree-widget/src/components/trees/stateless/models-tree/internal/ModelsTreeVisibilityHandler.ts

```diff
@@ -143,7 +187,19 @@ class ModelsTreeVisibilityHandlerImpl implements ModelsTreeVisibilityHandler {
   }
 
   private getVisibilityStatusObs(node: HierarchyNode): Observable<VisibilityStatus> {
+    if (node.filtering?.filteredChildrenIdentifierPaths?.length && !node.filtering.isFilterTarget) {
+      return this.getFilteredNodeVisibility({
+        parentKeys: [...node.parentKeys, node.key],
+        filterPaths: node.filtering.filteredChildrenIdentifierPaths,
+      });
+    }
+
     if (HierarchyNode.isClassGroupingNode(node)) {
+      const ancestor = node.nonGroupingAncestor;
+      if (ancestor && HierarchyNode.isInstancesNode(ancestor) && ancestor.filtering?.filteredChildrenIdentifierPaths) {
+        return this.getVisibilityStatusObs(ancestor);
+      }
+
       return this.getClassGroupingNodeDisplayStatus(node);
     }
```

Comment:
Come to think of it, we don't need to change the behavior for grouping nodes, because grouped instance ids are already filtered. Removed this change.